test/Psy/Test/VersionUpdater/CheckerTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy\Test\VersionUpdater;
+
+use Psy\Shell;
+use Psy\VersionUpdater\Checker;
+
+class CheckerTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider malformedResults
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage Unable to check for updates
+     * @param $input
+     */
+    public function testExceptionInvocation($input)
+    {
+        $checker = $this->getMockBuilder(Checker::class)
+            ->setMethods(['downloadFromGitHub'])
+            ->getMock();
+        $checker->expects($this->once())->method('downloadFromGitHub')->willReturn($input);
+        $checker->isLatest();
+    }
+
+    /**
+     * @dataProvider jsonResults
+     * @param $assertion
+     * @param $input
+     */
+    public function testDataSetResults($assertion, $input)
+    {
+        $checker = $this->getMockBuilder(Checker::class)
+            ->setMethods(['downloadFromGitHub'])
+            ->getMock();
+        $checker->expects($this->once())->method('downloadFromGitHub')->willReturn($input);
+        $this->assertSame($assertion, $checker->isLatest());
+    }
+
+    /**
+     * @return array
+     */
+    public function jsonResults()
+    {
+        return [
+            [false, json_decode('{"tag_name":"v9.0.0"}')],
```

Comment:
shorthand array syntax doesn't work in PHP 5.3

side note: if someone wants to make an argument for dropping support for older PHP, this would be way more compelling to me than "it's deprecated so we should force users to upgrade" :P